pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -3010,6 +3011,15 @@ public void shouldParseOutsideExprAtTimeZoneExtension() {
     Assert.assertEquals(fun.operands.get(0).getFunctionCall().operands.get(1).getLiteral().getStringValue(), "pst");
   }
 
+  @Test
+  public void testExtractTableNamesFromNode() {
```

Comment:
Can you add some more queries here (particularly join queries)? Examples:

```
SELECT COUNT(*) FROM tbl1 WHERE userUUID IN (SELECT userUUID FROM tbl2);
// add a few more cases of this with a mix of IN/Not-In, etc.
```

Some explicit join queries:

```
SELECT A.col1, B.col2 FROM tbl1 AS A JOIN tbl2 AS B ON A.key = B.key WHERE...
```

Some queries which use sub-queries (would be interesting to see if the tmp table name is also returned here):

```
with tmp as (
  select col1, count(*) from tbl1 where .. group by col1
)
select sum(col1) from tmp.
```